### PR TITLE
Added transformAlignment for container

### LINKED ISSUE #29875: Add `transformAlignment` parameter to `Container` widget.
When I use `Container` with `transform` parameter, I found that `Matrix4` always transforms widget depend on the origin(0, 0) ,but the `Transform` widget (which `Container` wraps, simply passed the `transform` argument to the `Transform` widget.) has a `alignment` parameter to decide the anchor to perform the transformation.

```
[✓] Flutter (Channel beta, v1.3.8, on Mac OS X 10.14.3 18D109, locale en-CN)
    • Flutter version 1.3.8 at /Users/yohom/Library/Flutter/flutter
    • Framework revision e5b1ed7a7f (3 weeks ago), 2019-03-06 14:23:37 -0800
    • Engine revision f4951df193
    • Dart version 2.2.1 (build 2.2.1-dev.0.0 571ea80e11)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/yohom/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /Users/yohom/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    • ios-deploy 1.9.4
    • CocoaPods version 1.6.0.beta.2

[✓] Android Studio (version 3.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 33.4.1
    • Dart plugin version 182.5215
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[✓] IntelliJ IDEA Ultimate Edition (version 2018.3.5)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 33.3.2
    • Dart plugin version 183.5912.23

[✓] VS Code (version 1.32.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.24.0
```

### PR DESCRIPTION
**Pull request by Andy-Python-Programmer**
## Description
Added transfromAlignment parameter to container

## Related Issues
Fixes #29875

## Tests

I added the following tests:

I have added tests for the transformAlignment parameter in tests/container_test.dart

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

### CODE REVIEW

**Comment by PixelToast** [comment]
Previous PR was #65916 which had an accidental rebase

File: packages/flutter/lib/src/widgets/container.dart [hunk 1]
```diff
@@ -356,6 +357,9 @@ class Container extends StatelessWidget {
   /// The transformation matrix to apply before painting the container.
   final Matrix4? transform;
 
+  /// /// The transform alignment for the transfrom property.
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/container.dart:360]: remove the extra `/// `

Comment (goderbauer) [inline packages/flutter/lib/src/widgets/container.dart:360]: This doc comment needs more details describing what this alignment is and also that it is only used when [transform] is not null.

File: packages/flutter/test/widgets/container_test.dart [hunk 2]
```diff
@@ -470,6 +470,47 @@ void main() {
...
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+
+    final RenderBox box = tester.renderObject(find.byKey(key));
+    expect(box.size, equals(const Size(100, 100)));
+    expect(box.localToGlobal(Offset.zero), equals(const Offset(100, 100)));
```
Comment (goderbauer) [inline packages/flutter/test/widgets/container_test.dart:511]: You could use tester.getSize and `tester.getTopLeft` here to make the test slightly more readable.

File: packages/flutter/lib/src/widgets/container.dart [hunk 3]
```diff
@@ -356,6 +357,10 @@ class Container extends StatelessWidget {
   /// The transformation matrix to apply before painting the container.
   final Matrix4? transform;
 
+  /// The transform alignment of the origin, relative to the size of the container.
+  /// Only applied when [transform] is **not** null.
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/container.dart:361]: How about:

```suggestion
 /// The alignment of the origin, relative to the size of the container, if [transform] is specified.
 ///
 /// When [transform] is null, the value of this property is ignored.
 ///
 /// See also:
 ///
 ///  * [Transform.align], which is set by this property.
```

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**